File: 1009/main.cpp
class Solution
{
public:
    int bitwiseComplement(int n) {
        int complement = 0;
        int offsetBits = 1;

        do {
            complement += n & offsetBits ? 0 : offsetBits;
            offsetBits <<= 1;
        } while (n > offsetBits);

        return complement;
    }
};

File: 1009/main.py
class Solution:
    def bitwiseComplement(self, n: int) -> int:
        if n == 0:
            return 1

        complement = 0
        shiftBit = 1

        while n > shiftBit:
            complement += 0 if n & shiftBit else shiftBit
            shiftBit <<= 1

        return complement
